babel/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -692,3 +692,80 @@ SqlNode DateTimeTerm() :
         )
     )
 }
+
+// This excludes CompoundIdentifier() as a data type.
+SqlDataTypeSpec DataTypeAlternativeCastSyntax() :
+{
+    SqlTypeNameSpec typeName;
+    final Span s;
+}
+{
+    typeName = TypeNameAlternativeCastSyntax() {
+        s = span();
+    }
+    (
+        typeName = CollectionsTypeName(typeName)
+    )*
+    {
+        return new SqlDataTypeSpec(
+            typeName,
+            s.end(this));
+    }
+}
+
+// This excludes CompoundIdentifier() as a type name that's found in the
+// original TypeName() function. Custom data types can be parsed
+// in parser.dataTypeParserMethods.
+SqlTypeNameSpec TypeNameAlternativeCastSyntax() :
+{
+    final SqlTypeNameSpec typeNameSpec;
+    final SqlIdentifier typeName;
+    final Span s = Span.of();
+}
+{
+    (
+<#-- additional types are included here -->
+<#-- put custom data types in front of Calcite core data types -->
+<#list parser.dataTypeParserMethods as method>
+        LOOKAHEAD(2)
```

Comment:
Isn't Babel's default LOOKAHEAD 2? If so, is this (and the one below) needed?

Reply:
This comment is moot if we go with the other suggestion.

Reply:
Yeah I'm not sure why they had that in the original TypeName method if that's the default for babel, but I agree with the other suggestion to add a parameter